theano/tensor/signal/downsample.py

```diff
@@ -1,1088 +1,57 @@
-                    c_st -= %(pd1)s;
-                    c_end -= %(pd1)s;
-                    // the maximum value
-                    maximum = ((dtype_%(z)s*)(PyArray_GETPTR4(%(z)s,b,k,i,j)))[0];
-                    // the gradient corresponding to this maximum value in z
-                    dtype_%(gz)s * gz = (
-                          (dtype_%(gz)s*)(PyArray_GETPTR4(%(gz)s, b, k, i, j)));
-                    // go through the pooled region in the unpadded input
-                    for(int m=r_st; m<r_end; m++)
-                    {
-                      for(int n=c_st; n<c_end; n++)
-                      {
-                        dtype_%(x)s a = ((dtype_%(x)s*)(PyArray_GETPTR4(%(x)s,b,k,m,n)))[0];
-                        dtype_%(gx)s * gx = (
-                          (dtype_%(gx)s*)(PyArray_GETPTR4(%(gx)s, b, k, m, n)));
-                        if (a == maximum){
-                          gx[0] = gx[0] + gz[0];
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            }
-
-        }
-        """ % locals()
-
-    def c_code_cache_version(self):
-        return (0, 7)
-
-DownsampleFactorMaxGrad = MaxPoolGrad
-
-
-class AveragePoolGrad(PoolGrad):
-
-    def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='average_inc_pad'):
-        assert mode in ['sum', 'average_inc_pad', 'average_exc_pad']
-        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode)
-
-    def make_node(self, x, gz):
-        # make_node should only be called by the grad function of
-        # DownsampleFactorMax, so these asserts should not fail.
-        assert isinstance(x, Variable) and x.ndim == 4
-        assert isinstance(gz, Variable) and gz.ndim == 4
-        x = tensor.as_tensor_variable(x)
-        gz = tensor.as_tensor_variable(gz)
-
-        return Apply(self, [x, gz], [x.type()])
-
-    def perform(self, node, inp, out):
-        if self.mode == 'average_exc_pad' and self.padding != (0, 0):
-            raise NotImplementedError()
-        x, gz = inp
-        gx_stg, = out
-        z_shape = self.out_shape(x.shape, self.ds, self.ignore_border, self.st,
-                                 self.padding)
-        if (gx_stg[0] is None) or (gx_stg[0].shape != z_shape):
-            gx_stg[0] = numpy.empty(z_shape, dtype=x.dtype)
-        zz = gx_stg[0]
-        # number of pooling output rows
-        pr = zz.shape[-2]
-        # number of pooling output cols
-        pc = zz.shape[-1]
-        ds0, ds1 = self.ds
-        st0, st1 = self.st
-        pad_h = self.padding[0]
-        pad_w = self.padding[1]
-        img_rows = x.shape[-2] + 2 * pad_h
-        img_cols = x.shape[-1] + 2 * pad_w
-        inc_pad = self.mode == 'average_inc_pad'
-        sum_mode = self.mode == 'sum'
+    warnings.warn("function 'max_pool2D' is now located "
+                  "in 'pool'.")
+    return pool.max_pool2D(*args, **kwargs)
```

Comment:
Also make the old class name available here, to allow unpickling them. Example:

DownsampleFactorMax = Pool

Also raise a warning when this file is imported and tell people the should import the pool file.